src/System.Private.ServiceModel/tests/Common/Infrastructure/CertificateManager.cs

```diff
@@ -104,6 +122,41 @@ internal static StoreLocation PlatformSpecificRootStoreLocation
             return resultCert;
         }
 
+        // Adds the given certificate to the given keychain unless it is
+        // already present.  Returns the  certificate either already in
+        // the store or the one requested.
+        public static X509Certificate2 AddToOSXKeyChainIfNeeded(SafeKeychainHandle keychain,
+                                                          X509Certificate2 certificate)
```

Comment:
Seems it's only used to install cert to a custom keychain, could we rename the method to something like AddToOSXCustomKeyChainIfNeeded?

Reply:
I don't understand, the method is already called `AddToOSXKeyChainIfNeeded`?

Reply:
There is a system KeyChain on OSX and this method is not for system KeyChain. I want to reflect it in the method name.

Reply:
You can create an instance of SafeKeychainHandle for the system keychain and use this method if you want and it will work.